AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1377,6 +1408,19 @@ - (void)recursivelyClearFetchedData
 - (void)layout
 {
   ASDisplayNodeAssertMainThread();
+  
+  if (!_flags.isMeasured) {
```

Comment:
Class that implements manual layout should have a complete implementation of -layout and should not call this default implementation. 

Class that uses automatic layout can safely rely on this method. (However, it can still override -layout and call [super layout] if it wants to add a custom layout logic, which is a very interesting idea in itself. Example can be found in KittenNode)